RingCluster[] rings
ClusterInCamera[] inCamera
time stamp